source/common/network/address_impl.cc

```diff
@@ -171,6 +192,7 @@ Ipv4Instance::Ipv4Instance(absl::Status&, const sockaddr_in* address,
   friendly_name_.append(ip_.friendly_address_);
   friendly_name_.push_back(':');
   friendly_name_.append(port.data(), port.size());
+  error = validateProtocolSupported();
```

Comment:
do this first and early-exit if failed?

also please factor out all copied blocks of code into helpers.

Reply:
also please factor out all copied blocks of code. Scan over your PRs for all new versions of constructors and factor out common blocks into init() methods you can call from each variant.

Reply:
These ctors actually don't have much common pattern to be factored out, they are slightly different.